### PR TITLE
[ci] simplify R CI scripts

### PR DESCRIPTION
**Pull request by jameslamb**
I've been learning about [`shellcheck`](https://github.com/koalaman/shellcheck), a linter for shell scripts.

This PR has a small simplification to `test_r_package.sh` that it suggested: replacing `grep "something | wc -l` with `grep --count "something"`. This might make that script a tiny tiny bit faster, but more importantly makes it a little easier to understand.

This PR also simplifies away a duplicated line in `test_r_package_windows.ps1` that I noticed while working on #3523

### CODE REVIEW

File: .ci/test_r_package.sh [hunk 1]
```diff
@@ -222,8 +221,7 @@ fi
 if [[ $R_BUILD_TYPE == "cran" ]]; then
     pragma_warning_present=$(
         cat $BUILD_LOG_FILE \
-        | grep -E "warning: unknown pragma ignored" \
-        | wc -l
+        | grep --count -E "warning: unknown pragma ignored"
```
Comment (StrikerRUS) [inline .ci/test_r_package.sh:224]: Can we apply the same for
https://github.com/microsoft/LightGBM/blob/5dbd1df4b62a8fd33d93fe0a3c5f1bb9c3660e7c/.github/workflows/r_package.yml#L190
?

Comment (jameslamb) [reply .ci/test_r_package.sh:224]: oh yep!

Comment (jameslamb) [reply .ci/test_r_package.sh:224]: updated in https://github.com/microsoft/LightGBM/pull/3535/commits/cdabc9784234486563a41d2ce43aaf5c9521bf34

**Review by StrikerRUS — APPROVED** [review]
Thanks! LGTM!

**Pull request merged**